BaseClasses.py

```diff
@@ -563,9 +563,20 @@ def all_done():
         return False
 
 
+PathValue = Tuple[str, Optional["PathValue"]]
+
+
 class CollectionState():
-    additional_init_functions: List[Callable] = []
-    additional_copy_functions: List[Callable] = []
+    prog_items: _Counter[Tuple[str, int]]
```

Comment:
i think `import typing` and `prog_items: typing.Counter[...]` would be easier on the eyes than `import as`.